rs/execution_environment/src/hypervisor/tests.rs

```diff
@@ -2182,6 +2182,185 @@ fn ic0_call_cycles_add_has_no_effect_without_ic0_call_perform() {
     );
 }
 
+#[test]
```

Comment:
nit: For these tests maybe leave a couple of comments explaining the expectations. E.g. for the first one, we should see that the entire amount is added while in the second it would only add up to what the freezing threshold dictates.